chainstate/minercraft_init.go

```diff
@@ -0,0 +1,181 @@
+package chainstate
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/BuxOrg/bux/utils"
+	"github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/tonicpow/go-minercraft/v2"
+	"github.com/tonicpow/go-minercraft/v2/apis/mapi"
+)
+
+func (c *Client) minercraftInit(ctx context.Context) error {
+	if txn := newrelic.FromContext(ctx); txn != nil {
+		defer txn.StartSegment("start_minercraft").End()
+	}
+	mi := &minercraftInitializer{client: c, ctx: ctx, minersWithFee: make(minerToFeeMap)}
+
+	if err := mi.newClient(); err != nil {
+		return err
+	}
+
+	if err := mi.validateMiners(); err != nil {
+		return err
+	}
+
+	if c.isFeeQuotesEnabled() {
+		c.options.config.feeUnit = mi.lowestFee()
+	}
+
+	return nil
+}
+
+type minercraftInitializer struct {
+	client        *Client
+	ctx           context.Context
+	minersWithFee minerToFeeMap
+	lock          sync.Mutex
+}
+
+type (
+	minerID       string
+	minerToFeeMap map[minerID]utils.FeeUnit
+)
+
+func (i *minercraftInitializer) defaultMinercraftOptions() (opts *minercraft.ClientOptions) {
+	c := i.client
+	opts = minercraft.DefaultClientOptions()
+	if len(c.options.userAgent) > 0 {
+		opts.UserAgent = c.options.userAgent
+	}
+	return
+}
+
+func (i *minercraftInitializer) newClient() (err error) {
+	c := i.client
+	// No client?
+	if c.Minercraft() == nil {
+		var optionalMiners []*minercraft.Miner
+		var loadedMiners []string
+
+		// Loop all broadcast miners and append to the list of miners
+		for i := range c.options.config.minercraftConfig.broadcastMiners {
+			if !utils.StringInSlice(c.options.config.minercraftConfig.broadcastMiners[i].MinerID, loadedMiners) {
+				optionalMiners = append(optionalMiners, c.options.config.minercraftConfig.broadcastMiners[i])
+				loadedMiners = append(loadedMiners, c.options.config.minercraftConfig.broadcastMiners[i].MinerID)
+			}
+		}
+
+		// Loop all query miners and append to the list of miners
+		for i := range c.options.config.minercraftConfig.queryMiners {
```

Comment:
Like for broadcasting, can you use 
```suggestion
		for _, queryMiner := range c.options.config.minercraftConfig.queryMiners {
```